janus_server/src/task.rs

```diff
@@ -51,64 +47,35 @@ pub enum VdafInstance {
     FakeFailsPrepStep,
 }
 
-/// An HMAC-SHA-256 key used to authenticate messages exchanged between
-/// aggregators. See `agg_auth_key` in draft-gpew-priv-ppm §4.2.
-// We define the type this way because while we can use `ring::hmac::Key::new`
-// to get a `ring::hmac::Key` from a slice of bytes, we can't get the bytes
-// back out of the key.
-#[derive(Clone, Debug)]
-pub struct AggregatorAuthKey {
-    bytes: Vec<u8>,
-    hmac_key: hmac::Key,
-}
-
-// TODO(brandon): use a ring constant once one is exposed. This is the correct value per ring:
-//   https://docs.rs/ring/0.16.20/src/ring/digest.rs.html#339
-// (but we can't use the value in ring as a const because it's not const, it's static)
-const SHA256_BLOCK_LEN: usize = 512 / 8;
-
-impl AggregatorAuthKey {
-    pub fn new(key_bytes: &[u8]) -> Result<Self, Error> {
-        if key_bytes.len() < SHA256_OUTPUT_LEN || key_bytes.len() > SHA256_BLOCK_LEN {
-            return Err(Error::AggregatorAuthKeySize);
-        }
-        Ok(Self {
-            bytes: Vec::from(key_bytes),
-            hmac_key: hmac::Key::new(HMAC_SHA256, key_bytes),
-        })
-    }
+/// An authentication (bearer) token used by aggregators for aggregator-to-aggregator
+/// authentication.
+#[derive(Clone)]
+pub struct AggregatorAuthenticationToken(Vec<u8>);
 
-    /// Randomly generate an [`AggregatorAuthKey`].
-    pub fn generate() -> Self {
-        let mut key_bytes = [0u8; SHA256_BLOCK_LEN];
-        thread_rng().fill(&mut key_bytes);
-        // Won't panic: key_bytes is constructed as SHA256_BLOCK_LEN bytes, which will pass the
-        // validation check.
-        Self::new(&key_bytes).unwrap()
+impl From<Vec<u8>> for AggregatorAuthenticationToken {
+    fn from(token: Vec<u8>) -> Self {
+        Self(token)
     }
 }
 
-impl AsRef<[u8]> for AggregatorAuthKey {
-    fn as_ref(&self) -> &[u8] {
-        &self.bytes
+impl AggregatorAuthenticationToken {
+    /// Returns a view of the aggregator authentication token as a byte slice.
+    pub(crate) fn as_bytes(&self) -> &[u8] {
+        &self.0
     }
 }
 
-impl AsRef<hmac::Key> for AggregatorAuthKey {
-    fn as_ref(&self) -> &hmac::Key {
-        &self.hmac_key
-    }
-}
-
-impl PartialEq for AggregatorAuthKey {
+impl PartialEq for AggregatorAuthenticationToken {
     fn eq(&self, other: &Self) -> bool {
-        // The key is ignored because it is derived from the key bytes.
-        // (also, ring::hmac::Key doesn't implement PartialEq)
-        self.bytes == other.bytes
+        // We attempt constant-time comparisons of the token data. Note that this function still
+        // leaks whether the lengths of the tokens are equal -- this is acceptable because we expect
+        // the length of the tokens to provide enough randomness that needs to be guessed even if
```

Comment:
Fixing a typo in this comment:
```suggestion
        // the contents of the tokens to provide enough randomness that needs to be guessed even if
```